### ISSUE TITLE
Bookmarks do not update references when pages are moved/deleted

### ISSUE THREAD

**Issue by grdekhayser** [ddvk/remarkable-hacks#179]
To reproduce- 
- Create a notebook with 5 pages. 
- Bookmark page #5, confirm that works- click Bookmark icon, shows "5", goes there ok.
- Delete page #3 (or any page prior to #5).
- Click Bookmark icon- STILL SHOWS "5", even though that page is now page "4". 

Side effects - 
- cannot delete that "5" bookmark until you add enough pages to make up that #. 
- After delete, if there ARE enough pages, the bookmark now refers to whatever page is now the number. 

Workaround-
- Prior to deleting/moving pages around, one must delete all bookmarks in a notebook. (Ugh)

**Comment by ddvk** [comment]
the bookmarks were meant for pdfs, were the number of pages does not change. they point to numbers and are not linked to a page in anyway.

what can be improved is a delete button in the bookmark edit dialog (activated by long press on the bookmark in the menu)

**Comment by DCS-87** [comment]
But couldnt this be an easy to implement improvement on the bookmarks? I use them for sections in a notebook as well, and have to remove and add a page later for all following bookmarks everytime I add a page to a section. Just a thought!

**Comment by yeancek** [comment]
yes. I agree with DCS-87. I am using my RM2 for language learning also. I created a note file for vocabulary, labelling pages from A to Z. After bookmarking all pages, I added empty pages between alphabets for new words to be added, with a hope that it would make it easy to toggle between alphabets while learning. 

Unfortunately, I found out that the bookmarks do not shift together with the page where it was set.

Therefore, a big favour to ask you, ddvk, if you could improve bookmark function. It would be of immense help. Thanks in advanced.
